main.tf

```diff
@@ -214,6 +240,29 @@ data "aws_iam_policy_document" "default" {
 #Description : Terraform module to create Elasticsearch policy resource on AWS.
 resource "aws_elasticsearch_domain_policy" "default" {
   count           = var.enabled ? 1 : 0
-  domain_name     = var.domain_name
+  domain_name     = var.domain_name != "" ? var.domain_name : module.labels.id
   access_policies = join("", data.aws_iam_policy_document.default.*.json)
 }
+
+#Module      : ROUTE53
+#Description : Provides a Route53 record resource.
+resource "aws_route53_record" "es" {
+  count   = var.dns_enabled ? 1 : 0
+  zone_id = var.dns_zone_id
+  name    = var.es_hostname
+  type    = var.type
```

Comment:
this is wrong, we should use, route 53 module

Reply:
Yes, I'll update